internal/controller/server/client_services.go

```diff
@@ -0,0 +1,76 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"octavius/internal/controller/server/execution"
+	"octavius/internal/pkg/idgen"
+	"octavius/internal/pkg/log"
+
+	octerr "octavius/internal/pkg/errors"
+	clientCPproto "octavius/internal/pkg/protofiles/client_CP"
+)
+
+type customCTXKey string
+
+var uidKey customCTXKey = "uid"
+
+type clientCPServicesServer struct {
+	procExec execution.Execution
+}
+
+// NewProcServiceServer used to create a new execution context
+func NewProcServiceServer(exec execution.Execution) clientCPproto.ClientCPServicesServer {
+	return &clientCPServicesServer{
+		procExec: exec,
+	}
+}
+
+func (s *clientCPServicesServer) PostMetadata(ctx context.Context, request *clientCPproto.RequestToPostMetadata) (*clientCPproto.MetadataName, error) {
+	uid, err := idgen.NextID()
+	if err != nil {
+		log.Error(err, "error while assigning id to the request")
```

Comment:
this log is unnecessary. is this the top-level function that will send GRPC response to the client? I think let the client receive the error instead of logged it here.

return the error here. because this will caused the flow to still continue to below.

Reply:
if we are not able to assign ID to request so should we return back to client or just log it in server and proceed?

Reply:
we just logged it here on the server because it appears that error in assigning a UID should not be stop execution.

Reply:
Okay, so presently we did return the error and end the execution, but I guess this is something i need more info

Reply:
Is ID will be stored in etcd? If yes I don't think we should continue the flow.

Reply:
So I check the code and looks like that uid is not used anywhere in the flow. I think we can remove the uid assignment part. and put the code back when we need it.

Reply:
The UID is being assigned to every request so that we can track the logs for every one of them. Also, as of now we are not storing them in etcd. @jasoet  asked us to implement them for logging use case only.

Reply:
The uid generated helps to keep track of all the logs that happens against it by creating a Request Context. It is just for logging purpose as suggested by @jasoet

Reply:
Let's take note. if this only for logging purposes I suggest adding a feature to log library to add id in every log.